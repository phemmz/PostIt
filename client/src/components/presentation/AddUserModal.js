import React, { Component } from 'react';
import ReactDOM from 'react-dom';
/**
 * AddUserModal class
 */
class AddUserModal extends Component {
	render() {
		return (
			<div id="addUser" className="modal">
        <div className="modal-content">
          <h4 className="green-text text-darken-4">Add User to a Group</h4>
          <div className="row">
            <div className="input-field col s12">
              <select ref="dropdown" id="username" value={this.props.value} onChange={this.props.onChange} style={{display: 'block'}}>
                <option value="" disabled>Add User</option>
                {/* Populate the select dropdown with registered users  */}
                {this.props.appUsers}
              </select> 
            </div>
          </div>
        </div>
        <div className="modal-footer">
          <a className="modal-action modal-close waves-effect waves-green btn-flat">Cancel</a>
          <a onClick={this.props.onClick} className="modal-action modal-close waves-effect waves-green btn-flat">Add User</a>
        </div>
      </div>
		)
	}
}
export default AddUserModal;