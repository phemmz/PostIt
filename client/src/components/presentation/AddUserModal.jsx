import React from 'react';
import PropTypes from 'prop-types';
import Chips from 'react-chips';

const AddUserModal = ({
  onClick, users, suggestions,
  onChipsChange, addUserSuccess, addUserFail,
  existingUser, usersAddedSuccessfully
}) => {
  return (
    <div id="addUser" className="modal">
      <div className="modal-content">
        <h4 className="green-text text-darken-4">Add Users to a Group</h4>
        <hr />
        <div className="row">
          {
            (addUserSuccess) ?
            (
              <div
                className="col s5 add-user-success green-text center"
              >
                {usersAddedSuccessfully.join()} Successfully Added to Group
              </div>
            ) :
            null
          }
          {
            (addUserFail) ?
            (
              <div
                className="col s5 add-user-fail red-text center"
              >
                {existingUser.join()} Already Exist In The Group
              </div>
            ) :
            null
          }
        </div>
        <div className="row">
          <div className="input-field col s12">
            <Chips
              id="users"
              value={users}
              onChange={onChipsChange}
              suggestions={suggestions}
              placeholder="Add users to a group"
            />
          </div>
        </div>
      </div>
      <div className="modal-footer">
        <a
          className="modal-action modal-close waves-effect waves-green btn-flat"
        >Cancel</a>
        <a
          onClick={onClick}
          className="modal-action waves-effect waves-green btn-flat add-btn"
          role="link"
          tabIndex={0}
        >Add User</a>
      </div>
    </div>
  );
};

AddUserModal.propTypes = {
  onClick: PropTypes.func.isRequired,
  users: PropTypes.arrayOf(PropTypes.string).isRequired,
  suggestions: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChipsChange: PropTypes.func.isRequired,
  addUserSuccess: PropTypes.bool.isRequired,
  addUserFail: PropTypes.bool.isRequired,
  existingUser: PropTypes.arrayOf(PropTypes.string),
  usersAddedSuccessfully: PropTypes.arrayOf(PropTypes.string)
};

AddUserModal.defaultProps = {
  existingUser: [],
  usersAddedSuccessfully: []
};

export default AddUserModal;
