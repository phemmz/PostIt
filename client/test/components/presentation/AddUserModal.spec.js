import React from 'react';
import { mount } from 'enzyme';
import AddUserModal from '../../../src/components/presentation/AddUserModal';
import { suggestions,
   updateUser, groupDetails } from '../../__mockData__/dummyProps';

describe('AddUserModal Component', () => {
  let users = ['phemmz'];
  const onClick = jest.fn();
  const onChipsChange = jest.fn();
  const addUserSuccess = false;
  const addUserFail = false;
  const wrapper = mount(
    <AddUserModal
      users={users}
      onChange={updateUser}
      suggestions={suggestions}
      onClick={onClick}
      onChipsChange={onChipsChange}
      addUserSuccess={addUserSuccess}
      addUserFail={addUserFail}
    />);
  it('should always render a div', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').exists()).toBe(true);
  });
  
  it('should call onClick function when the Add User link is clicked',
  () => {
    wrapper.find('.add-btn').simulate('click');
    expect(onClick.mock.calls.length).toBe(1);
  });

  it('should receive props', () => {
    expect(Object.keys(wrapper.props()).length).toBe(9);
  });
});
