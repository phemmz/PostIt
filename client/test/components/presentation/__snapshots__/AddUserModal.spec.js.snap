// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUserModal Component should always render a div 1`] = `
<AddUserModal
  addUserFail={false}
  addUserSuccess={false}
  existingUser={Array []}
  onChipsChange={[Function]}
  onClick={[Function]}
  suggestions={
    Array [
      "phemmz",
      "nawtuaw",
    ]
  }
  users={
    Array [
      "phemmz",
    ]
  }
  usersAddedSuccessfully={Array []}
>
  <div
    className="modal"
    id="addUser"
  >
    <div
      className="modal-content"
    >
      <h4
        className="green-text text-darken-4"
      >
        Add Users to a Group
      </h4>
      <hr />
      <div
        className="row"
      />
      <div
        className="row"
      >
        <div
          className="input-field col s12"
        >
          <Chips
            createChipKeys={
              Array [
                9,
              ]
            }
            fetchSuggestions={null}
            fetchSuggestionsThrushold={10}
            fromSuggestionsOnly={false}
            getChipValue={[Function]}
            getSuggestionValue={[Function]}
            id="users"
            onChange={[Function]}
            placeholder="Add users to a group"
            renderChip={[Function]}
            renderLoading={[Function]}
            renderSuggestion={[Function]}
            suggestions={
              Array [
                "phemmz",
                "nawtuaw",
              ]
            }
            suggestionsFilter={[Function]}
            theme={
              Object {
                "chipsContainer": Object {
                  ":focus": Object {
                    "border": "1px solid #aaa",
                  },
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "border": "1px solid #ccc",
                  "borderRadius": 5,
                  "display": "flex",
                  "flexWrap": "wrap",
                  "font": "13.33333px Arial",
                  "minHeight": 24,
                  "padding": "2.5px",
                  "position": "relative",
                },
                "container": Object {
                  "flex": 1,
                },
                "containerOpen": Object {},
                "input": Object {
                  "border": "none",
                  "boxSizing": "border-box",
                  "margin": 2.5,
                  "outline": "none",
                  "padding": 5,
                  "width": "100%",
                },
                "sectionContainer": Object {},
                "sectionTitle": Object {},
                "suggestion": Object {
                  "padding": "5px 15px",
                },
                "suggestionHighlighted": Object {
                  "background": "#ddd",
                },
                "suggestionsContainer": Object {},
                "suggestionsList": Object {
                  "backgroundColor": "#fff",
                  "border": "1px solid #ccc",
                  "left": 0,
                  "listStyle": "none",
                  "margin": 0,
                  "padding": 0,
                  "position": "absolute",
                  "top": "100%",
                  "width": "100%",
                  "zIndex": 10,
                },
              }
            }
            uniqueChips={true}
            value={
              Array [
                "phemmz",
              ]
            }
          >
            <div
              data-radium={true}
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "border": "1px solid #ccc",
                  "borderRadius": 5,
                  "display": "flex",
                  "flexWrap": "wrap",
                  "font": "13.33333px Arial",
                  "minHeight": 24,
                  "padding": "2.5px",
                  "position": "relative",
                }
              }
            >
              <Chip
                index={0}
                onRemove={[Function]}
                selected={false}
                theme={
                  Object {
                    "chip": Object {
                      "background": "#ccc",
                      "borderRadius": 3,
                      "cursor": "default",
                      "margin": "2.5px",
                      "padding": 5,
                    },
                    "chipRemove": Object {
                      ":hover": Object {
                        "color": "red",
                      },
                      "cursor": "pointer",
                      "fontWeight": "bold",
                    },
                    "chipSelected": Object {
                      "background": "#888",
                    },
                  }
                }
              >
                <div
                  data-radium={true}
                  style={
                    Object {
                      "background": "#ccc",
                      "borderRadius": 3,
                      "cursor": "default",
                      "margin": "2.5px",
                      "padding": 5,
                    }
                  }
                >
                  phemmz
                  <span
                    data-radium={true}
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                        "fontWeight": "bold",
                      }
                    }
                  >
                     Ã—
                  </span>
                </div>
              </Chip>
              <Autosuggest
                alwaysRenderSuggestions={false}
                createChipKeys={
                  Array [
                    9,
                  ]
                }
                fetchSuggestions={null}
                fetchSuggestionsThrushold={10}
                focusInputOnSuggestionClick={true}
                fromSuggestionsOnly={false}
                getChipValue={[Function]}
                getSuggestionValue={[Function]}
                highlightFirstSuggestion={false}
                id="users"
                inputProps={
                  Object {
                    "onBlur": [Function],
                    "onChange": [Function],
                    "onFocus": [Function],
                    "onKeyDown": [Function],
                    "placeholder": "Add users to a group",
                    "value": "",
                  }
                }
                multiSection={false}
                onChange={[Function]}
                onSuggestionSelected={[Function]}
                onSuggestionsClearRequested={[Function]}
                onSuggestionsFetchRequested={[Function]}
                placeholder="Add users to a group"
                renderChip={[Function]}
                renderLoading={[Function]}
                renderSuggestion={[Function]}
                renderSuggestionsContainer={[Function]}
                shouldRenderSuggestions={[Function]}
                suggestions={Array []}
                suggestionsFilter={[Function]}
                theme={
                  Object {
                    "chipsContainer": Object {
                      ":focus": Object {
                        "border": "1px solid #aaa",
                      },
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "border": "1px solid #ccc",
                      "borderRadius": 5,
                      "display": "flex",
                      "flexWrap": "wrap",
                      "font": "13.33333px Arial",
                      "minHeight": 24,
                      "padding": "2.5px",
                      "position": "relative",
                    },
                    "container": Object {
                      "flex": 1,
                    },
                    "containerOpen": Object {},
                    "input": Object {
                      "border": "none",
                      "boxSizing": "border-box",
                      "margin": 2.5,
                      "outline": "none",
                      "padding": 5,
                      "width": "100%",
                    },
                    "sectionContainer": Object {},
                    "sectionTitle": Object {},
                    "suggestion": Object {
                      "padding": "5px 15px",
                    },
                    "suggestionHighlighted": Object {
                      "background": "#ddd",
                    },
                    "suggestionsContainer": Object {},
                    "suggestionsList": Object {
                      "backgroundColor": "#fff",
                      "border": "1px solid #ccc",
                      "left": 0,
                      "listStyle": "none",
                      "margin": 0,
                      "padding": 0,
                      "position": "absolute",
                      "top": "100%",
                      "width": "100%",
                      "zIndex": 10,
                    },
                  }
                }
                uniqueChips={true}
                value={
                  Array [
                    "phemmz",
                  ]
                }
              >
                <Autowhatever
                  getSectionItems={[Function]}
                  highlightedItemIndex={null}
                  highlightedSectionIndex={null}
                  id="users"
                  inputProps={
                    Object {
                      "onBlur": [Function],
                      "onChange": [Function],
                      "onFocus": [Function],
                      "onKeyDown": [Function],
                      "placeholder": "Add users to a group",
                      "value": "",
                    }
                  }
                  itemProps={[Function]}
                  items={Array []}
                  multiSection={false}
                  renderInputComponent={[Function]}
                  renderItem={[Function]}
                  renderItemData={
                    Object {
                      "query": "",
                    }
                  }
                  renderItemsContainer={[Function]}
                  renderSectionTitle={[Function]}
                  theme={
                    Object {
                      "chipsContainer": Object {
                        ":focus": Object {
                          "border": "1px solid #aaa",
                        },
                        "alignItems": "center",
                        "backgroundColor": "#fff",
                        "border": "1px solid #ccc",
                        "borderRadius": 5,
                        "display": "flex",
                        "flexWrap": "wrap",
                        "font": "13.33333px Arial",
                        "minHeight": 24,
                        "padding": "2.5px",
                        "position": "relative",
                      },
                      "container": Object {
                        "flex": 1,
                      },
                      "containerOpen": Object {},
                      "input": Object {
                        "border": "none",
                        "boxSizing": "border-box",
                        "margin": 2.5,
                        "outline": "none",
                        "padding": 5,
                        "width": "100%",
                      },
                      "item": Object {
                        "padding": "5px 15px",
                      },
                      "itemHighlighted": Object {
                        "background": "#ddd",
                      },
                      "itemsContainer": Object {},
                      "itemsList": Object {
                        "backgroundColor": "#fff",
                        "border": "1px solid #ccc",
                        "left": 0,
                        "listStyle": "none",
                        "margin": 0,
                        "padding": 0,
                        "position": "absolute",
                        "top": "100%",
                        "width": "100%",
                        "zIndex": 10,
                      },
                      "sectionContainer": Object {},
                      "sectionTitle": Object {},
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <input
                      aria-activedescendant={null}
                      aria-autocomplete="list"
                      aria-expanded={false}
                      aria-haspopup={false}
                      aria-owns="react-autowhatever-users"
                      autoComplete="off"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Add users to a group"
                      role="combobox"
                      style={
                        Object {
                          "border": "none",
                          "boxSizing": "border-box",
                          "margin": 2.5,
                          "outline": "none",
                          "padding": 5,
                          "width": "100%",
                        }
                      }
                      type="text"
                      value=""
                    />
                    <div
                      id="react-autowhatever-users"
                      style={Object {}}
                    />
                  </div>
                </Autowhatever>
              </Autosuggest>
            </div>
          </Chips>
        </div>
      </div>
    </div>
    <div
      className="modal-footer"
    >
      <a
        className="modal-action modal-close waves-effect waves-green btn-flat"
      >
        Cancel
      </a>
      <a
        className="modal-action waves-effect waves-green btn-flat add-btn"
        onClick={[Function]}
        role="link"
        tabIndex={0}
      >
        Add User
      </a>
    </div>
  </div>
</AddUserModal>
`;
